xarray/core/variable.py

```diff
@@ -932,11 +936,7 @@ def copy(self, deep=True, data=None):
                 # don't share caching between copies
                 data = indexing.MemoryCachedArray(data.array)
 
-            if deep and (
-                hasattr(data, "__array_function__")
-                or isinstance(data, dask_array_type)
-                or (not IS_NEP18_ACTIVE and isinstance(data, np.ndarray))
-            ):
+            if deep and is_duck_array(data):
                 data = copy.deepcopy(data)
```

Comment:
then we'd have something like this?
```suggestion
            if deep:
                data = copy.deepcopy(data)
```

Reply:
Yes, exactly! This is definitely much closer to what I would expect `copy(deep=True)` to do without looking at the implementation.